chain/network/src/announce_accounts/mod.rs

```diff
@@ -0,0 +1,108 @@
+use crate::store;
+use lru::LruCache;
+use near_primitives::network::{AnnounceAccount, PeerId};
+use near_primitives::types::AccountId;
+use parking_lot::Mutex;
+use std::collections::HashMap;
+
+#[cfg(test)]
+mod tests;
+
+const ANNOUNCE_ACCOUNT_CACHE_SIZE: usize = 10_000;
+
+struct Inner {
+    /// Maps an account_id to a peer owning it.
+    account_peers: LruCache<AccountId, AnnounceAccount>,
+    /// Subset of account_peers, which we have broadcasted to the peers.
+    /// It is used to skip rebroadcasting the same data multiple times.
+    /// It contains less entries than account_peers in case some AnnounceAccounts
+    /// have been loaded from storage without broadcasting.
+    account_peers_broadcasted: LruCache<AccountId, AnnounceAccount>,
+    /// Access to store on disk
+    store: store::Store,
+}
+
+impl Inner {
+    /// Get AnnounceAccount for the given AccountId.
+    fn get_announce(&mut self, account_id: &AccountId) -> Option<AnnounceAccount> {
+        if let Some(aa) = self.account_peers.get(account_id) {
+            return Some(aa.clone());
+        }
+        match self.store.get_account_announcement(&account_id) {
+            Err(e) => {
+                tracing::warn!(target: "network", "Error loading announce account from store: {:?}", e);
+                None
+            }
+            Ok(None) => None,
+            Ok(Some(a)) => {
+                self.account_peers.put(account_id.clone(), a.clone());
+                Some(a)
+            }
+        }
+    }
+}
+
+pub(crate) struct Cache(Mutex<Inner>);
```

Comment:
nit: Cache is a bit too generic name for a public struct, how about AccountAnnounceCache? 
feel free to prove me wrong with rust guidelines ;)

Reply:
I think you are right. We have a similar pattern here: https://github.com/near/nearcore/pull/9086/files#diff-f8335427ec06c1343d68539896edbfe231bf67164af9bf8bacc17b4db00aea1aR105. We can write `use crate::accounts_data;` then refer to it as `accounts_data::Cache` which is cute. But `AccountsDataCache` and `AccountAnnounceCache` seem like better choices overall.

Reply:
Sent a separate PR for `AccountsDataCache`: https://github.com/near/nearcore/pull/9129